test/cli_run_dns_test.go

```diff
@@ -66,3 +68,72 @@ func (suite *PouchRunSuite) TestRunWithBridgeNetwork(c *check.C) {
 
 	c.Assert(res.Stdout(), check.Equals, hostRes.Stdout())
 }
+
+// TestRunWithDNSFlags tests DNS related flags.
+func (suite *PouchRunDNSSuite) TestRunWithDNSFlags(c *check.C) {
+	cname := "TestRunWithDNSFlags"
+
+	res := command.PouchRun("run", "--name", cname,
+		"--dns", "1.2.3.4",
+		"--dns-option", "timeout:3",
+		"--dns-search", "example.com",
+		busyboxImage,
+		"cat", "/etc/resolv.conf")
+	defer DelContainerForceMultyTime(c, cname)
+	res.Assert(c, icmd.Success)
+
+	// test if the value is in inspect result
+	inspectRes := command.PouchRun("inspect", cname)
+	res.Assert(c, icmd.Success)
+
+	result := []types.ContainerJSON{}
+	if err := json.Unmarshal([]byte(inspectRes.Stdout()), &result); err != nil {
+		c.Errorf("failed to decode inspect output: %v", err)
+	}
+
+	c.Assert(result[0].HostConfig.DNS, check.DeepEquals, []string{"1.2.3.4"})
+	c.Assert(result[0].HostConfig.DNSOptions, check.DeepEquals, []string{"timeout:3"})
+	c.Assert(result[0].HostConfig.DNSSearch, check.DeepEquals, []string{"example.com"})
+
+	// test if the value is correct in container
+	out := strings.Trim(res.Stdout(), "\n")
+	c.Assert(strings.Contains(out, "nameserver 1.2.3.4"), check.Equals, true)
+	c.Assert(strings.Contains(out, "options timeout:3"), check.Equals, true)
+	c.Assert(strings.Contains(out, "search example.com"), check.Equals, true)
+}
+
+// TestRunWithDNSRepeatFlags tests repeated DNS related flags.
+func (suite *PouchRunDNSSuite) TestRunWithDNSRepeatFlags(c *check.C) {
+	cname := "TestRunWithDNSRepeatFlags"
+
+	res := command.PouchRun("run", "--name", cname,
+		"--dns", "1.2.3.4",
+		"--dns", "2.3.4.5",
+		"--dns-option", "timeout:3",
+		"--dns-option", "ndots:9",
+		"--dns-search", "mydomain",
+		"--dns-search", "mydomain2",
+		busyboxImage,
+		"cat", "/etc/resolv.conf")
+	defer DelContainerForceMultyTime(c, cname)
+	res.Assert(c, icmd.Success)
```

Comment:
It's a good chioce to put `command.PouchRun` and `Assert` into one line? Since the res var didn't use after this.

Reply:
> @ZYecho Fixed.

LGTM, but waiting for the CI pass.